bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/communication/SonnenJSONCommunication.java

```diff
@@ -38,21 +39,64 @@ public class SonnenJSONCommunication {
 
     private Gson gson;
     private @Nullable SonnenJsonDataDTO batteryData;
+    private SonnenJsonPowerMeterDataDTO @Nullable [] powerMeterData;
 
     public SonnenJSONCommunication() {
         gson = new Gson();
         config = new SonnenConfiguration();
     }
 
     /**
-     * Refreshes the battery connection.
+     * Refreshes the battery connection with the new API Call V2.
      *
      * @return an empty string if no error occurred, the error message otherwise.
      */
-    public String refreshBatteryConnection() {
+    public String refreshBatteryConnectionAPICALLV2(boolean powerMeter) {
         String result = "";
-        String urlStr = "http://" + config.hostIP + "/api/v1/status";
+        String urlStr = "http://" + config.hostIP + "/api/v2/status";
+        Properties httpHeader = new Properties();
+        httpHeader = createHeader(config.authToken);
+        try {
+            String response = HttpUtil.executeUrl("GET", urlStr, httpHeader, null, "application/json", 10000);
+            logger.debug("BatteryData = {}", response);
+            if (response == null) {
+                throw new IOException("HttpUtil.executeUrl returned null");
+            }
+            batteryData = gson.fromJson(response, SonnenJsonDataDTO.class);
+
+            if (powerMeter) {
+                response = HttpUtil.executeUrl("GET", "http://" + config.hostIP + "/api/v2/powermeter", httpHeader,
+                        null, "application/json", 10000);
+                logger.debug("PowerMeterData = {}", response);
+                if (response == null) {
+                    throw new IOException("HttpUtil.executeUrl returned null");
+                }
+
+                powerMeterData = gson.fromJson(response, SonnenJsonPowerMeterDataDTO[].class);
+            }
+        } catch (IOException | JsonSyntaxException e) {
+            logger.debug("Error processiong Get request {}:  {}", urlStr, e.getMessage());
+
+            if (e.getMessage() != null && e.getMessage().contains("WWW-Authenticate header")) {
```

Comment:
A compiler warning is introduced here. It can be fixed by:
```suggestion
            String message = e.getMessage();
            if (message != null && message.contains("WWW-Authenticate header")) {
```

Reply:
ok